pkg/controller/daemon/daemoncontroller.go

```diff
@@ -1105,6 +1129,12 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 		return false, false, false, err
 	}
 
+	// Return directly if there are intential predicates reasons. This can make sure daemoncontroller will not
+	// fire InsufficientResourceError if there are intential predicate reasons.
```

Comment:
daemonset controller didn't fire this error, but emit an event. Also, `InsufficientResourceError` isn't the only case that's skipped. There's typo for "intentional". Update the comment to something like:

```go
// Return directly if there is any intentional predicate failure reason, so that daemonset controller skips
// checking other predicate failures, such as InsufficientResourceError and unintentional errors.
```